test/archiveTest.js

```diff
@@ -500,182 +498,6 @@ describe('case archive tracking', () => {
-          expect(isValidDate(res.body[1].filesLoadedIntoVidarrArchival)).to.be
-            .true;
-          expect(res.body[0].filesLoadedIntoVidarrArchival).to.equal(
-            existingTime
-          );
-          expect(res.body[1].filesLoadedIntoVidarrArchival).not.to.equal(
-            existingTime
-          );
-          done();
-        });
-      });
-    });
-    it('it should update only one of two archive entries for a given case identifier (files sent offsite)', (done) => {
-      let caseIdentifier = 'R13_TEST_9999_De_F';
-      getCaseByCaseIdentifier(server, caseIdentifier).end((err, res) => {
-        expect(res.status).to.equal(200);
-        let existingID = res.body[0].commvaultBackupJobId;
-        expect(existingID).not.to.be.a('null');
-        expect(res.body[1].commvaultBackupJobId).to.be.a('null');
-        let reqBody = {
-          commvaultBackupJobId: 'JC1313',
-        };
-        updateCaseArchives(
-          server,
-          caseIdentifier,
-          urls.filesSentOffsite,
-          reqBody
-        ).end((err, res) => {
-          expect(res.status).to.equal(200);
-          expect(res.body[1].commvaultBackupJobId).not.to.be.a('null');
-          expect(res.body[1].commvaultBackupJobId).to.equal(
-            reqBody.commvaultBackupJobId
-          );
-          expect(res.body[0].commvaultBackupJobId).to.equal(existingID);
-          done();
-        });
-      });
-    });
-    it('it should create a new archive record when one is completed for a given case identifier', (done) => {
-      let reqBody = {
-        caseIdentifier: 'R12_TEST_1212_Ab_C',
-        requisitionId: 12,
-        limsIds: [
-          '901_1_LDI9001',
-          '901_1_LDI9002',
-          '902_1_LDI9001',
-          '902_1_LDI9002',
-        ],
-        workflowRunIdsForOffsiteArchive: [
-          'vidarr:research/run/qwerty',
-          'vidarr:research/run/9876',
-        ],
-        workflowRunIdsForVidarrArchival: ['vidarr:research/run/abba'],
-      };
-      addCaseArchives(server, reqBody).end((err, res) => {
-        expect(res.status).to.equal(201);
-        done();
-      });
-    });
-    it('it should not create a new archive record when one is completed and reqisition is different', (done) => {
-      let reqBody = {
-        caseIdentifier: 'R12_TEST_1212_Ab_C',
-        requisitionId: 22,
-        limsIds: [
-          '901_1_LDI9001',
-          '901_1_LDI9002',
-          '902_1_LDI9001',
-          '902_1_LDI9002',
-        ],
-        workflowRunIdsForOffsiteArchive: [
-          'vidarr:research/run/asdf',
-          'vidarr:research/run/1234',
-        ],
-        workflowRunIdsForVidarrArchival: ['vidarr:research/run/abba'],
-      };
-      addCaseArchives(server, reqBody).end((err, res) => {
-        expect(res.status).to.equal(409);
-        done();
-      });
-    });
-    it('it should not create a new archive record when one is completed and lims IDs are different', (done) => {
```

Comment:
It may be good to keep this test.

Additional test cases to consider:
* posting the same case+lims ids+archive files should return success, but not update or create a new archive
* posting the same case id but different lims id or archive files should fail, and not update the existing case archive or create a new case archive

Reply:
I think the tests at line 117 and 152 cover these two scenarios.